### PR TITLE
Fix animation options karma test

### LINKED ISSUE #6953: Fix Animation Options Karma Tests
2 animation tests are failing occasionally due to `ResizeObserver loop limit exceeded thrown`. The tests are currently disabled. it should be fixed and enabled. 

- plays the animation when a control in the panel is changed.
- replaces an existing effect with a new one

### PR DESCRIPTION
**Pull request by embarks**
## Context
This PR re-enables a flaky test and introduces some fixes to help the test pass. The test was for checking that the animation plays after the effect chooser is changed. The other fixes are detailed in Relevant Technical Choices.

## Summary
- The error which was causing the test to fail reads as follows: 
- `ResizeObserver loop limit exceeded thrown`
- This was due to a "benign error" with the ResizeObserver, some discussion can be found [here in a github issue thread for the resize-observer](https://github.com/WICG/resize-observer/issues/38#issuecomment-422126006) and [here-- in the stack overflow thread which details the chosen solution.](https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded)

- The error is intermittent. I ran the karma test suite 10 times on `main` and saw a failure once. In order to test I found where the resize observer was being instantiated when interacting with the effect chooser dropdown and running and added the `requestAnimationFrame` there which prevents the error from occurring.
- In order to verify, I re-ran the same test suite 30 times and never saw the error locally.

## Relevant Technical Choices
I noticed a couple other problems while I was replicating:
- animations not firing in Firefox when animation dropdown is changed
    - I added 200ms to the debounce callback which seemed to resolve this (And a 300ms sleep to the karma test as a result of that change)
- console errors when None selected 
    - I added a return to prevent these errors

## User-facing changes
This is a bug fix.

### QA

- [x] This is a non-user-facing change and requires no QA

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6953

### CODE REVIEW

**Comment by swissspidy** [comment]
> * This was due to a "benign error" with the ResizeObserver, some discussion can be found [here in a github issue thread for the resize-observer](https://github.com/WICG/resize-observer/issues/38#issuecomment-422126006) and [here-- in the stack overflow thread which details the chosen solution.](https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded)

Looks like we already do the same in `TextPane`. Might be time to document this somewhere, otherwise every dev ends up doing the same research at some point 😄 

In other places we just ignore the errors (as most people seem to do)

Also https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded#comment118281231_58701523 is worth reading.

**Comment by BrittanyIRL** [comment]
> Looks like we already do the same in `TextPane`. Might be time to document this somewhere, otherwise every dev ends up doing the same research at some point 😄

I think if anything is the outcome of last week's bummer of having to disable a bunch of tests it's this - we really should start documenting this better.

**Comment by embarks** [comment]
> > * This was due to a "benign error" with the ResizeObserver, some discussion can be found [here in a github issue thread for the resize-observer](https://github.com/WICG/resize-observer/issues/38#issuecomment-422126006) and [here-- in the stack overflow thread which details the chosen solution.](https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded)
> 
> Looks like we already do the same in `TextPane`. Might be time to document this somewhere, otherwise every dev ends up doing the same research at some point 😄
> 
> In other places we just ignore the errors (as most people seem to do)
> 
> Also https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded#comment118281231_58701523 is worth reading.

I'll add some documentation! 

I saw the cypress fix, which is nice, but I'm not sure how to add an event handler for uncaught exceptions for jest/karma... I did some searching and I'm not sure if it runs on `process` or what the global is called to add a handler in the same way. Are you familiar with that @swissspidy?

**Review by samwhale — APPROVED** [review]
Nice, this works locally for me. Would be great if we could get a document of common 'gotchas' to track stuff like this

File: assets/src/edit-story/components/panels/design/animation/karma/animation.karma.js [hunk 1]
```diff
@@ -75,9 +75,7 @@ describe('Animation Panel', function () {
     expect(effectChooser.innerText).toBe('Drop');
   });
 
-  // TODO #6953
-  // eslint-disable-next-line jasmine/no-disabled-tests
-  xit('plays the animation when a control in the panel is changed.', async function () {
+  it('plays the animation when a control in the panel is changed.', async function () {
```
Comment (samwhale) [inline assets/src/edit-story/components/panels/design/animation/karma/animation.karma.js:79]: There's another test right above this that I think you can unskip too!

**Review by BrittanyIRL — APPROVED** [review]
Also works locally for me!

**Pull request merged**